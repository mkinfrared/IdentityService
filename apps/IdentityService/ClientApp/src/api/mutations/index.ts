export * from "./loginMutation";

export * from "./registerMutation";

export * from "./verifyEmailMutation";

export * from "./submitConsent";

export * from "./submitForgotPassword";

export * from "./submitPasswordChange";
