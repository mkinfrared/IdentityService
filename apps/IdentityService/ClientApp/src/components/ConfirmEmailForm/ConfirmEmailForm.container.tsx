import { memo } from "react";

import { useReturnUrl } from "hooks";

import ConfirmEmailFormFC from "./ConfirmEmailForm";
import { ConfirmEmailFormContainerProps } from "./ConfirmEmailForm.type";

const ConfirmEmailFormContainer = (props: ConfirmEmailFormContainerProps) => {
  const returnUrl = useReturnUrl();

  return <ConfirmEmailFormFC {...props} returnUrl={returnUrl} />;
};

export default memo(ConfirmEmailFormContainer);
