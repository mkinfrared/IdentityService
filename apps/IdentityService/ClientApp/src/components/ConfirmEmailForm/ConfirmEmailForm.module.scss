@import "../../styles/common";

.ConfirmEmailForm {
  @extend %login-register-form;
}

.Story {
}
