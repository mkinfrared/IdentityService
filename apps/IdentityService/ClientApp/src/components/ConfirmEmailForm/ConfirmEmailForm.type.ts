export type ConfirmEmailFormProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  returnUrl: string;
};

export type ConfirmEmailFormData = {
  code: string;
};
