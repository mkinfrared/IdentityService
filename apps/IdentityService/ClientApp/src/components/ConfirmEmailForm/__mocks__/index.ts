export { default } from "./ConfirmEmailForm.mock";
