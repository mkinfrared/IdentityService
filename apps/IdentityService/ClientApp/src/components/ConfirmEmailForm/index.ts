export { default } from "./ConfirmEmailForm.container";

export * from "./ConfirmEmailForm.type";
