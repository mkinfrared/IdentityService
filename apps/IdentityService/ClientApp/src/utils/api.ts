import axios from "axios";

import { BASE_URL } from "utils/secrets";

const { isAxiosError } = axios;

const api = axios.create({
  baseURL: BASE_URL,
  withCredentials: true,
});

export { isAxiosError };

export default api;
