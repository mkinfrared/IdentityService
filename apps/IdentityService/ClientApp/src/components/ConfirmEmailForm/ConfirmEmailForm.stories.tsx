import { Meta, Story } from "@storybook/react";

import { ConfirmEmailForm } from "./ConfirmEmailForm";
import { ConfirmEmailFormProps } from "./ConfirmEmailForm.type";
import css from "./Story.module.scss";

export default {
  title: "COMPONENTS/ConfirmEmailForm",
  component: ConfirmEmailForm,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<ConfirmEmailFormProps> = (args) => (
  <div className={css.Story}>
    <ConfirmEmailForm {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
