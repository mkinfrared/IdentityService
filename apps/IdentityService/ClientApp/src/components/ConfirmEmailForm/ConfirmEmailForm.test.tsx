import { fireEvent, render, waitFor } from "@testing-library/react";

import * as mutations from "api/mutations";

import { ConfirmEmailForm } from "./ConfirmEmailForm";

jest.mock("api/mutations");

describe("<ConfirmEmailForm />", () => {
  const { sessionStorage, location } = window;
  const returnUrl = "https://foobar.com/";
  const Component = <ConfirmEmailForm returnUrl={returnUrl} />;
  const mutationsMock = mutations as jest.Mocked<typeof mutations>;

  beforeAll(() => {
    Object.defineProperty(window, "sessionStorage", {
      configurable: true,
      value: {
        ...sessionStorage,
        getItem: jest.fn(),
      },
    });

    Object.defineProperty(window, "location", {
      configurable: true,
      value: {
        ...location,
        assign: jest.fn(),
      },
    });
  });

  afterAll(() => {
    Object.defineProperty(window, "sessionStorage", {
      configurable: true,
      value: sessionStorage,
    });

    Object.defineProperty(window, "location", {
      configurable: true,
      value: location,
    });
  });

  it("should be defined", () => {
    expect(ConfirmEmailForm).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should disable submit button if code is empty", () => {
    const { container, getByTestId } = render(Component);
    const codeField = container.querySelector('input[name="code"]');
    const submitButton = getByTestId("Button");

    fireEvent.change(codeField!, { target: { value: "" } });

    expect(submitButton).toBeDisabled();
  });

  it("should call verifyEmailMutation on form submit and redirect to returnUrl if form is valid", async () => {
    const getItem = jest.spyOn(window.sessionStorage, "getItem");

    getItem.mockReturnValue("marklar");

    const { container, getByTestId } = render(Component);
    const codeField = container.querySelector('input[name="code"]');
    const submitButton = getByTestId("Button");
    const button = getByTestId("Button");

    expect(submitButton).toBeDisabled();

    fireEvent.click(button);

    fireEvent.change(codeField!, { target: { value: "424242" } });

    await waitFor(() => expect(submitButton).toBeEnabled());

    await fireEvent.submit(submitButton);

    await waitFor(() =>
      expect(mutationsMock.verifyEmailMutation).toHaveBeenCalled(),
    );
  });

  it("should throw an error if token or user ID are missing in session storage", async () => {
    const getItem = jest.spyOn(window.sessionStorage, "getItem");

    getItem.mockReturnValue(null);

    const { container, getByTestId } = render(Component);
    const codeField = container.querySelector('input[name="code"]');
    const submitButton = getByTestId("Button");
    const button = getByTestId("Button");

    expect(submitButton).toBeDisabled();

    fireEvent.click(button);

    fireEvent.change(codeField!, { target: { value: "424242" } });

    await waitFor(() => expect(submitButton).toBeEnabled());

    await fireEvent.submit(submitButton);

    await waitFor(() =>
      expect(mutationsMock.verifyEmailMutation).not.toHaveBeenCalled(),
    );

    expect(window.location.assign).not.toHaveBeenCalled();
  });
});
