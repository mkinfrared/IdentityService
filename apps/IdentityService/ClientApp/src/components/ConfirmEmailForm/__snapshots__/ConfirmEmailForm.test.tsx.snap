// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmEmailForm /> should match the snapshot 1`] = `
<div>
  <div
    class="ConfirmEmailForm"
  >
    <form
      class="form"
    >
      <svg
        class="icon"
      >
        email-icon.svg
      </svg>
      <h3
        class="Heading-module_Heading__Wg8yB heading"
        data-testid="Heading"
      >
        Check your email
      </h3>
      <label
        class="TextField-module_TextField__QtlJl"
        data-testid="TextField"
      >
        <div
          class="TextField-module_input__SPywc"
        >
          <input
            autocomplete="off"
            name="code"
            type="text"
            value=""
          />
        </div>
        <p
          class="Text-module_Text__K8mLZ TextField-module_error__IGQzp"
          data-testid="Text"
        />
      </label>
      <div
        class="buttonWrapper"
      >
        <button
          class="Button-module_Button__0OJEu Button-module_opaque__O0DEF"
          data-testid="Button"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
